alicloud/resource_alicloud_cs_kubernetes.go

```diff
@@ -475,9 +505,31 @@ func resourceAlicloudCSKubernetesRead(d *schema.ResourceData, meta interface{})
 	d.Set("master_disk_category", cluster.Parameters.MasterSystemDiskCategory)
 	d.Set("worker_disk_size", cluster.Parameters.WorkerSystemDiskSize)
 	d.Set("worker_disk_category", cluster.Parameters.WorkerSystemDiskCategory)
-	d.Set("worker_data_disk_size", cluster.Parameters.WorkerDataDiskSize)
-	d.Set("worker_data_disk_category", cluster.Parameters.WorkerDataDiskCategory)
 	d.Set("availability_zone", cluster.ZoneId)
+	d.Set("node_cidr_mask", cluster.Parameters.NodeCIDRMask)
+
+	if cluster.Parameters.WorkerDataDisk {
+		d.Set("worker_data_disk_size", cluster.Parameters.WorkerDataDiskSize)
+		d.Set("worker_data_disk_category", cluster.Parameters.WorkerDataDiskCategory)
+	}
+
+	if cluster.Parameters.LoggingType != "" {
+		logConfig := map[string]interface{}{}
+
+		if cluster.Parameters.LoggingType == "None" {
+			logConfig["type"] = ""
+		} else {
+			logConfig["type"] = cluster.Parameters.LoggingType
+			if cluster.Parameters.SLSProjectName == "None" {
+				logConfig["sls_project_name"] = ""
+			} else {
+				logConfig["sls_project_name"] = cluster.Parameters.SLSProjectName
+			}
+			if err := d.Set("log_config", []map[string]interface{}{logConfig}); err != nil {
+				return err
+			}
```

Comment:
I think if LoggingType == "None" and SLSProjectName == "None", you can set log_config to a empty struct directly; otherwise set logtype and project.

Reply:
Conditions here do have issues.
I will set `type` when `type` is not None, no matter if `project` exists.